### ISSUE TITLE
signAndPost or signAndBroadcast / 404

### ISSUE THREAD

**Issue by okwme** [CosmWasm/cosmjs#439]
Following instructions from [@cosmjs/launchpad](https://cosmwasm.github.io/cosmjs/latest/launchpad/index.html) that says to construct and submit a transaction like this:
```js
import {
  coins,
  MsgSubmitProposal,
  OfflineSigner,
  PostTxResult,
  SigningCosmosClient,
  StdFee,
} from "@cosmjs/launchpad";

async function publishProposal(
  apiUrl: string,
  signer: OfflineSigner,
): Promise<PostTxResult> {
  const [{ address: authorAddress }] = await signer.getAccounts();

  const memo = "My first proposal on chain";
  const msg: MsgSubmitProposal = {
    type: "cosmos-sdk/MsgSubmitProposal",
    value: {
      content: {
        type: "cosmos-sdk/TextProposal",
        value: {
          description:
            "This proposal proposes to test whether this proposal passes",
          title: "Test Proposal",
        },
      },
      proposer: authorAddress,
      initial_deposit: coins(25000000, "ustake"),
    },
  };
  const fee: StdFee = {
    amount: coins(5000000, "ucosm"),
    gas: "89000000",
  };

  const client = new SigningCosmosClient(apiUrl, authorAddress, signer);
  return client.signAndPost([msg], fee, memo);
}
```
Except I'm just doing a simple  `MsgSend` like this (with gratuitous console logs):
```js
async function submitWithCosmJS(recipient) {
  const [{address}] = await signer.getAccounts();
  console.log({address})
  const memo = ''
  const msg = {
    type:  'cosmos-sdk/MsgSend',
    value: {
        amount:  [{ amount: '1', denom: 'nametoken' }],
        from_address: address,
        to_address: recipient
    }
  };
  const fee = {
    amount: coins(0, ''),
    gas: "200000",
  };
  console.log({restEndpoint})
  const client = new SigningCosmosClient(restEndpoint, address, signer);
  console.log({client})
  return client.signAndPost([msg], fee, memo);
}
```
~~I get a 404 as a response with the following content:~~
(removed because it was a misconfig on my node)

When I look in the docs [here](https://cosmwasm.github.io/cosmjs/latest/launchpad/classes/signingcosmosclient.html) for the `client` I don't see any record of `signAndPost()`, just `signAndBroadcast()` which I'm told `"client.signAndBroadcast is not a function"`.

Seems like there's two things here:
1. The docs are out of sync with the code
~2. The `signAndPost` method is crafting the `GET` request with an additional `index.htm` suffix~

Does this seem like a correct assessment?

**Comment by webmaster128** [comment]
The verb "post" [was renamed to](https://github.com/CosmWasm/cosmjs/issues/317) "broadcast" across all APIs between 0.22 and 0.23. If you want to use the latest APIs, you can use version `"^0.23.0-alpha.0` of all packages `@cosmjs/*`. Does that help your situation?

The docs are indeed out of sync since they are always built for the current master. That is not ideal.

**Comment by webmaster128** [comment]
The fee

```js
  const fee = {
    amount: coins(0, ''),
    gas: "200000",
  };
```

_might_ lead to an invalid signature. If that happens, try

```js
  const fee = {
    amount: [],
    gas: "200000",
  };
```

**Comment by webmaster128** [comment]
https://github.com/CosmWasm/cosmjs/pull/440 fixes a README bug in master / 0.23. Thanks for letting us know.

**Comment by okwme** [comment]
thanks for the help and quick fix!
